lib/ansible/cli/galaxy.py

```diff
@@ -1279,9 +1277,9 @@ def execute_info(self):
                 role_info.update(gr.metadata)
 
             req = RoleRequirement()
-            role_spec = req.role_yaml_parse({'role': role})
-            if role_spec:
-                role_info.update(role_spec)
+            if ',' in role:
+                raise AnsibleError("Invalid old style role requirement: %s" % role)
```

Comment:
are we removing the old way of specifying role in string? why mot remove from role_yaml_parse? deprecation?

Reply:
It keeps the exact behavior. All `role_yaml_parse` does if 'role' is in the dict, is validate there is no ',' in the role, and replace the key 'role' with 'name'. I thought this was a weird use of `role_yaml_parse` but I can put it back.

Reply:
no, this saves time and avoids the complexity of the full function, if that was the only reason we called it ... this is better.